sql/ballots/ballots.go

```diff
@@ -148,3 +148,28 @@ func CountByPubkeyLayer(db sql.Executor, lid types.LayerID, pubkey []byte) (int,
 	}
 	return rows, nil
 }
+
+// GetRefBallot gets a ref ballot for a layer and a pubkey.
+func GetRefBallot(db sql.Executor, epochID types.EpochID, pubkey []byte) (ballotID types.BallotID, err error) {
+	firstLayer := epochID.FirstLayer()
+	lastLayer := firstLayer.Add(types.GetLayersPerEpoch()).Sub(1)
+	rows, err := db.Exec(`
+		select * from ballots 
```

Comment:
in code you are consistent about getting ballot including the malicious field. i think you should do it here too.
i think the query should be the same as `Get()` with the extra where clause

Reply:
changed to `select id` instead